### ISSUE TITLE
doom-private-dir with symlinks don't have reduced flycheck errors

### ISSUE THREAD

**Issue by bricewge** [hlissner/doom-emacs#1550]
https://github.com/hlissner/doom-emacs/blob/72fd61e07bbce7599aa9a61a22c2ce7e35fbebf4/modules/lang/emacs-lisp/autoload.el#L127-L129
`+emacs-lisp|reduce-flycheck-errors-in-emacs-config` doesn't work when `doom-private-dir` contains symlink. When opening a file in `doom-private-dir` emacs follows the symlink so the reduce flycheck errors aren't applied.

**Comment by hlissner** [comment]
I can't reproduce this. My `~/.config/doom` is a symlink, and flycheck is correctly gimped in my private config. Could you elaborate on the structure of your config and what files are symlinked?

**Comment by bricewge** [comment]
Sure, the issue appear when opening any files in `doom-private-dir`, here is it's structure:
```
$ tree ~/.doom.d -L 1
/home/bricewge/.doom.d
├── config.el -> ../project/dotfiles/emacs/.doom.d/config.el
├── +eshell.el -> ../project/dotfiles/emacs/.doom.d/+eshell.el
├── init.el -> ../project/dotfiles/emacs/.doom.d/init.el
├── +org.el -> ../project/dotfiles/emacs/.doom.d/+org.el
├── packages.el -> ../project/dotfiles/emacs/.doom.d/packages.el
├── snippets
└── themes
```

**Comment by hlissner** [comment]
I don't think this is a problem.

It's simple to follow symlinks to determine where they go, but not the reverse. The only surefure way to detect this is to require an expensive and recursive crawl through `doom-private-dir` every time you open an *.el file to determine if the file exists in it, following a potentially indeterminable number of symlinks, which is impractical.

Doing so would also break the use case where external packages are symlinked into your DOOMDIR. Packages where you _would_ want syntax checking.

Can't you symlink `~/.doom.d` to `~/project/dotfiles/emacs/.doom.d` instead?

**Comment by bricewge** [comment]
Ok I see the problem here. When I opened the issue the only way I could imagine to solve it, in bad hackish way, was to add a variable to the list of dirs.

Unfortunately I can't just make `~/.doom.d` a symlink without deeply changing the way I manage my dotfiles. I would have to modify parameter passed to `sotw` on a per module basis to remove the options `--no-folding`. So I settled on adding a `.dir-locals.el` in my dotfiles' `.doom.d` containing:
```elisp
((emacs-lisp-mode
  (flycheck-disabled-checkers . 'emacs-lisp-checkdoc)))
```
It would help if the value of the local variable `flycheck-emacs-lisp-check-form` was accessible outside of `
+emacs-lisp|reduce-flycheck-errors-in-emacs-config` to add it in `.dir-locals.el` too.
